docs/api.rst

```diff
@@ -171,20 +171,45 @@ Replace each event value with x.::
 
   constant('tick', periodic(1000))
 
+.. _tap:
+
 tap
 ^^^
 
 .. code-block:: haskell
 
   tap :: (a -> *) -> Stream a -> Stream a
 
+Perform a side-effect for each event in stream.
+
+.. code-block:: javascript
+
+  stream:         -a-b-c-d->
+  tap(f, stream): -a-b-c-d->
+
+For each event in stream, f is called, but the value of its result is ignored. 
+If f fails (ie throws), then the returned stream will also fail. The stream 
+returned by tap will contain the same events as the original stream.
+
+.. _ap:
+
 ap
-^^
+^^^
 
 .. code-block:: haskell
 
   ap :: Stream (a -> b) -> Stream a -> Stream b
 
+Apply the latest function in a stream of functions to the latest value in stream.
+
+.. code-block:: javascript
+
+  streamOfFunctions:              --f---------g--------h------>
+  stream:                         -a-------b-------c-------d-->
+  ap(stream, streamOfFunctions.): --fa-----fb-gb---gc--hc--hd->
```

Comment:
This one is really hard to draw an ascii diagram for.  Do you think the `fa`, `fb`, etc stuff will make sense to readers?  I know it's what's in the most.js docs, but should we try to come up with something better, or do you think it's ok?